File: w03/d05/najee_gardner/calculator/lib/calculator.rb
class Calculator
  def add(num1, num2)
    num1 + num2
  end

  def subtract(num1, num2)
    num1 - num2
  end

  def sum(arr)
    return 0 if !arr.reduce(:+)
    arr.reduce(:+)

    ## McK solution
    # arr.reduce(0, :+)
  end

  def multiply(*nums)
    nums.reduce(:*)
  end

  def power(num, power)
    num**power

    ## Khadijah's answer (recursive)
    # power == 0 ? 1 : num * power(num, power - 1)
  end

  def factorial(num)
    num == 0 ? 1 : (1..num).reduce(:*)
  end
end

File: w06/d05/sean_jennings/calculator.js
function subtract (minuend,subtrahend) {

  if (minuend > subtrahend) {

    return (minuend - subtrahend);

  } else {

    return false;
  }

}
var answer = subtract(40,27);
console.log(answer);



function multiply (num1,num2) {
  product = 0;
  for (var i = num2; num2 > 0; num2--) {
    product = product + num1;
  }
  return product;
}
var product = multiply(5,4);

console.log(product);

function divideBy (numerator,denominator) {
  quotient = 0;
  for (var i = 0 ; numerator >= denominator; i++) {
    numerator = (numerator - denominator);
    quotient ++;
  }
return quotient;
}

var quotient = divideBy(50,45);
console.log(quotient);
